### PR TITLE
Fix various Cleanroom issues

### LINKED ISSUE #2123: Machines can't perform cleanroom recipes unless cleanroom was built after the machine was already placed
### Checked for existing issues

- [X] I have checked for existing issues, and have found none.

### Tested latest version

- [X] I have checked that this occurs on the latest version.

### GregTech CEu Version

1.4.4

### Minecraft Version

1.20.1

### Recipe Viewer Installed

EMI

### Environment

All

### Cross-Mod Interaction

No

### Other Installed Mods

EMI

### Expected Behavior

If you place a machine inside of a cleanroom, it should be able to perform recipes that require cleanroom environment.

### Actual Behavior

Machines are unable to perform cleanroom recipes until you break the cleanroom, and rebuild it.

Guessing placing machines in unfinished cleanroom and finishing the cleanroom after placing the machines also works, but I haven't tested that (should be the exact same scenario).

### Steps to Reproduce

1) Build a cleanroom
2) Place machines inside
3) Try to execute a recipe that require a cleanroom environment
4) Machine does nothing until the cleanroom is reformed

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by krossgg**
## What
Fixes #2123 - placing blocks now updates the Cleanroom properly
Fixes the one-cycle UI lag in the Cleanroom
Fixes Cleanroom allowing configurable/cleaning maintenance hatches

## Details
Added new ForgeHooks mixin to update multiblocks on block-place, as the ChunkMixin does not get called

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/core/mixins/ForgeHooksMixin.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(ForgeHooks.class)
+public class ForgeHooksMixin {
+
+    @Inject(method = "onPlaceItemIntoWorld",
+            at = @At(value = "INVOKE",
+                     opcode = Opcodes.GETFIELD,
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/core/mixins/ForgeHooksMixin.java]: remove this line

File: src/main/java/com/gregtechceu/gtceu/core/mixins/ForgeHooksMixin.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import net.minecraftforge.common.util.BlockSnapshot;
+
+import com.llamalad7.mixinextras.sugar.Local;
+import org.objectweb.asm.Opcodes;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(ForgeHooks.class)
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/core/mixins/ForgeHooksMixin.java]: add `remap=false`

File: src/main/java/com/gregtechceu/gtceu/core/mixins/ForgeHooksMixin.java (hunk 1 above)
Comment (screret) [reply src/main/java/com/gregtechceu/gtceu/core/mixins/ForgeHooksMixin.java]: also add `remap=true` to the `@At`

File: src/main/resources/gtceu.mixins.json
Comment (screret) [inline src/main/resources/gtceu.mixins.json:1]: I love it when the indentation changes on our JSONs

**Review by screret — DISMISSED** [review]

**Review by YoungOnionMC — APPROVED** [review]

**Pull request merged**